File: LeetCode/729-My-Calendar-I(Medium)/Solution.java
class MyCalendar {
    private List<int[]> calendar;

    public MyCalendar() {
        calendar = new ArrayList<>();
    }
    
    public boolean book(int start, int end) {
        for (int[] range : calendar)
            if (range[0] < end && start < range[1]) return false;
        
        calendar.add(new int[]{start, end});
        return true;
    }
}

/**
 * Your MyCalendar object will be instantiated and called as such:
 * MyCalendar obj = new MyCalendar();
 * boolean param_1 = obj.book(start,end);
 */

File: LeetCode/729-My-Calendar-I(Medium)/Solution.cpp
class MyCalendar {
private:
    vector<pair<int, int>> books;
public:
    MyCalendar() {

    }

    bool book(int start, int end) {
        for (auto book : books)
            if (max(book.first, start) < min(book.second, end))
                return false;
        books.push_back({start, end});

        return true;
    }
};

/**
 * Your MyCalendar object will be instantiated and called as such:
 * MyCalendar* obj = new MyCalendar();
 * bool param_1 = obj->book(start,end);
 */
